crates/red_knot/src/args.rs

```diff
@@ -71,6 +71,14 @@ pub(crate) struct CheckCommand {
     #[arg(long, value_name = "VERSION", alias = "target-version")]
     pub(crate) python_version: Option<PythonVersion>,
 
+    /// Target platform to assume when resolving types.
+    ///
+    /// This is used to specialize the type of `sys.platform` and will affect the visibility
+    /// of platform-specific functions and attributes. If the value is set to `all`, no
+    /// assumptions are made about the target platform.
+    #[arg(long, value_name = "PLATFORM", alias = "platform")]
```

Comment:
Both `mypy` and `pyright` call this CLI option `--platform`, so I thought it would be reasonable to add this alias. If we anticipate using `--platform` for something else potentially, we should remove it.